job-view.php

```diff
@@ -1,91 +1,122 @@
 $jobs_id = "";
-if (!isset($_GET['id']) || !is_numeric($_GET['id'])) {
-    header('Location: ' . BASE_URL);
+if ( ! isset( $_GET['id'] ) || ! is_numeric( $_GET['id'] ) ) {
+	header( 'Location: ' . BASE_URL );
 }
 
-$jobs_id = $_GET['id'];
-$sql = "SELECT jobs.*, user.company_name, user.company_site FROM jobs JOIN user ON user.id_user = jobs.id_user WHERE id_jobs = '$jobs_id'";
-$result = db_sql_run($sql);
+$jobs_id = mysqli_real_escape_string( open_db_conn(), $_GET['id'] );
+$sql     = "SELECT jobs.*, user.company_name, user.company_site FROM jobs JOIN user ON user.id_user = jobs.id_user WHERE id_jobs = '$jobs_id'";
+$result  = db_sql_run( $sql );
 
-if (mysqli_num_rows($result) < 1) {
-    header('Location: ' . BASE_URL);
+//redirect to the home page if there is no job matching such id in the database
+if ( mysqli_num_rows( $result ) < 1 ) {
+	header( 'Location: ' . BASE_URL );
 }
 
-while ($row = mysqli_fetch_assoc($result)) :
-    ?>
-    <section class="section-fullwidth">
-        <div class="row">
-            <div class="job-single">
-                <div class="job-main">
-                    <div class="job-card">
-                        <div class="job-primary">
-                            <header class="job-header">
-                                <h2 class="job-title"><?php echo $row['title'] ?></h2>
-                                <div class="job-meta">
-                                    <a class="meta-company" href="#"><?php echo $row['company_name'] ?></a>
-                                    <span class="meta-date">
+$row = mysqli_fetch_assoc( $result );
+
+
+$job = array(
+	'title'               => '',
+	'company_name'        => '',
+	'company_description' => '',
+	'company_site'        => '',
+	'publication_date'    => '',
+	'jobs_location'       => '',
+	'salary'              => '',
+	'description'         => '',
+	'id_user'             => '',
+);
+
+foreach ( $row as $key => $value ) {
+	if ( ! empty( $value ) && array_key_exists( $key, $job ) ) {
+		$job[ $key ] = $value;
+	}
+}
+
+
+//params for include template
+$meta_title = $job['title'] . " | Jobrix.tk";
+
+//header template include
+require_once( dirname( __FILE__ ) . '/includes/theme-compat/header.php' );
+?>
+<section class="section-fullwidth">
+    <div class="row">
+        <div class="job-single">
+            <div class="job-main">
+                <div class="job-card">
+                    <div class="job-primary">
+                        <header class="job-header">
+                            <h2 class="job-title"><?php echo $job['title'] ?></h2>
+                            <div class="job-meta">
+								<?php
+								if ( ! empty( $job['company_site'] ) ) { ?>
+                                    <a class="meta-company"
+                                       href="<?php echo 'http://' . $job['company_site'] ?>"><?php echo $job['company_name'] ?></a>
```

Comment:
the formatting of this is a bit weird, but what is more questionable is - why do we have `http://` and then we are getting `$job['company_site']`? What would happen if `$job['company_site']` is actually `https://devrix.com`?

Reply:
$job['company_site'] always returns a domain. It never returns a url. In the registration form, we ask the client to leave a site name: doman.com or www.doman.com